jdaviz/app.py

```diff
@@ -1347,6 +1347,43 @@ def merge_func(spectral_region):  # noop
             elif isinstance(subset_state, MultiMaskSubsetState):
                 return self._get_multi_mask_subset_definition(subset_state)
 
+    def delete_subsets(self, subset_labels=None):
+        """
+        Delete all or specified subsets in app.
+
+        This method removes subsets based on the provided ``subset_labels`` (a
+        single subset label or multiple labels). If ``subset_labels``
+        is None (default), all subsets will be removed.
+
+        Parameters
+        ----------
+        subset_labels : str or list of str or None
+            The label(s) of the subsets to delete. If None, all subsets will be
+            removed.
+        """
+
+        # delete all subsets
+        if subset_labels is None:
+            for subset_group in self.data_collection.subset_groups:
+                self.data_collection.remove_subset_group(subset_group)
+
+        else:
+            if not isinstance(subset_labels, (list, np.ndarray, tuple)):
+                subset_labels = [subset_labels]
+            labels = np.array(subset_labels)
+            sg = self.data_collection.subset_groups
+            subset_grp_labels = {sg[i].label: sg[i] for i in range(len(sg))}
```

Comment:
Simpler to access sg directly?
```suggestion
            subset_grp_labels = {s.label: s for s in sg}
```

Reply:
My comment here is marked as resolved but code was not accepted. Could you please clarify? Thanks!